InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs

```diff
@@ -0,0 +1,142 @@
+                {
+                    Barcode = ticket.Barcode,
+                    MemberId = 0
+                });
+            }
+
+            return attendanceList;
+        }
+
+        public List<Attendance> GetAttendanceListBySeasonId(int seasonId)
+        {
+            List<Product> products = service.Products().Where(x => x.SeasonId == seasonId).ToList();
+            List<Member> members = service.Members().ToList();
+            List<Ticket> tickets = service.Tickets().ToList();
+
+            List<Member> seasonMembers = new List<Member>();
+            List<Attendance> attendanceList = new List<Attendance>();
+
+            foreach (Product product in products)
+            {
+                foreach (Member member in members)
+                {
+                    if (member.ProductId == product.Id)
+                    {
+                        attendanceList.Add(new Attendance()
+                        {
+                            Barcode = "N/A",
+                            MemberId = member.Id
+                        });
+                    }
+                }
+            }
+
+            foreach (Product product in products)
+            {
+                foreach (Ticket ticket in tickets)
+                {
+                    if (ticket.ProductId == product.Id)
+                    {
+                        attendanceList.Add(new Attendance()
+                        {
+                            Barcode = ticket.Barcode,
+                            MemberId = 0
+                        });
+                    }
+                }
+            }
+
+
+            return attendanceList;
+        }
+
+        public List<Attendance> GetAttendanceListByGameId(int gameId)
+        {
+
+            List<Product> products = service.Products().Where(x => x.GameId == gameId).ToList();
+            List<Ticket> tickets = service.Tickets().ToList();
+
+            List<Attendance> attendanceList = new List<Attendance>();
+
+            foreach (Product product in products)
+            {
+                foreach (Ticket ticket in tickets)
+                {
+                    if (ticket.ProductId == product.Id)
+                    {
+                        attendanceList.Add(new Attendance()
+                        {
+                            Barcode = ticket.Barcode,
+                            MemberId = 0
+                        });
+                    }
+                }
+            }
+
+
+            return attendanceList;
+        }
+
+        public List<Attendance> GetAttendanceListBySeasonIdAndGameId(int seasonId, int gameId)
+        {
+            List<Ticket> tickets = service.Tickets().ToList();
+            List<Product> products = service.Products().Where(x => x.SeasonId == seasonId)
+                                                       .Where(x => x.GameId == gameId)
+                                                       .ToList();
```

Comment:
The collection of Products includes individual tickets (`Type = ProductType.Ticket`) with access to one game, and season passes (`Type = ProductType.Member`) with access to all games in a season, meaning they have access to the game specified here even though their GameId doesn't match (GameId is zero on all member products). Therefore, to get the right set of products you needed to do something like this:
```
var ticketProducts = service.Products().Where(p => p.SeasonId == seasonId && p.GameId == gameId)
var memberProducts = service.Products().Where(p => p.SeasonId == seasonId && p.Type == ProductType.Member);
```
Also, your `Where()` methods can be joined as I have shown above